### PR TITLE
Npm: Filter the file list already while generating it

### PR DESCRIPTION
**Pull request by sschuberth**
Prefer using FileFilter with listFiles() instead of first listing all
files and then filtering them later.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

### CODE REVIEW

File: analyzer/src/main/kotlin/managers/Npm.kt [hunk 1]
```diff
@@ -383,13 +384,13 @@ open class Npm(
         val nodeModulesDir = moduleDir.resolve("node_modules")
         if (!nodeModulesDir.isDirectory) return emptyList()
 
-        val searchDirs = nodeModulesDir.listFiles().filter { file ->
-            file.isDirectory && file.name.startsWith("@")
-        } + nodeModulesDir
+        val searchDirs = nodeModulesDir.listFiles(FileFilter {
```
Comment (fviernau) [inline analyzer/src/main/kotlin/managers/Npm.kt]: Consider keeping `file` instead of `it` for readability.

Comment (fviernau) [inline analyzer/src/main/kotlin/managers/Npm.kt]: commit-msg: generating **is** -> **it** ?

Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/Npm.kt]: Done.

Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/Npm.kt]: Actually, for these (almost) one-liners I personally prefer to keep it shorter with `it`, esp. as directly before it now says _**File**Filter_.

**Pull request merged**